src/Umbraco.Infrastructure/Media/ImageSharpImageUrlGenerator.cs

```diff
@@ -1,68 +1,88 @@
-﻿using System.Collections.Generic;
+using System;
+using System.Collections.Generic;
 using System.Globalization;
+using System.Linq;
 using System.Text;
+using SixLabors.ImageSharp;
 using Umbraco.Cms.Core.Media;
 using Umbraco.Cms.Core.Models;
-using Umbraco.Extensions;
 
 namespace Umbraco.Cms.Infrastructure.Media
 {
     public class ImageSharpImageUrlGenerator : IImageUrlGenerator
     {
-        public IEnumerable<string> SupportedImageFileTypes => new[] { "jpeg", "jpg", "gif", "bmp", "png" };
+        private static readonly string[] s_supportedImageFileTypes = Configuration.Default.ImageFormats.SelectMany(f => f.FileExtensions).ToArray();
+
+        public IEnumerable<string> SupportedImageFileTypes { get; } = s_supportedImageFileTypes;
```

Comment:
This should ensure all image extensions supported by ImageSharp are handled. I'm not sure if `Configuration.Default` reflects any custom added formats though (e.g. when WebP support is added as separate package). @JimBobSquarePants can you confirm this is the right way to get all supported image formats?

Reply:
Yep. They’re all loaded there by default

Reply:
This all worked fine whenever the ImageSharp.Web configuration used the same `Configuration.Default` instance (which always returns the same instance, because it's lazily initialized). If `ImageSharpMiddlewareOptions.Configuration` was set to a different instance, it would return the wrong image formats/file extensions.

I've updated the implementation to use the `IOptions` pattern, so it uses the same configuration as the middleware.

We'll probably have to do the same for `ImageDimensionExtractor`, as that uses `Image.Load(stream)` (without specifying the configuration and thus using the default).

@JimBobSquarePants Does ImageSharp support MS DI for fetching the default configuration? If so, we should probably register the configuration once and use that as default for both ImageSharp and ImageSharp.Web.

Reply:
I've now registered the default ImageSharp configuration and used that throughout Umbraco, so every implementation using ImageSharp can use that same instance (currently only `ImageDimensionExtractor` and the ImageSharp.Web middleware):
https://github.com/umbraco/Umbraco-CMS/blob/0b53249d5742404e7511d129ae55503ce8673c85/src/Umbraco.Infrastructure/DependencyInjection/UmbracoBuilder.CoreServices.cs#L182

As the supported image file types/extensions are fetched from this configuration, it's possible to only have PNG support using the following composer:
```c#
public class OnlyPngSupportComposer : IUserComposer
{
    public void Compose(IUmbracoBuilder builder) => builder.Services.AddUnique(new Configuration(new PngConfigurationModule()));
}
```